src/uu/dd/src/dd.rs

```diff
@@ -88,30 +88,105 @@ impl Num {
     }
 }
 
-struct Input<'a, R: Read> {
-    src: R,
+/// Data sources.
+enum Source {
+    /// Input from stdin.
+    Stdin(Stdin),
+
+    /// Input from a file.
+    File(File),
+
+    /// Input from a named pipe, also known as a FIFO.
+    #[cfg(any(target_os = "linux", target_os = "android"))]
```

Comment:
Is the problem not just that this should be Mac too?

Reply:
Or maybe it should be cfg(unix)?

Reply:
You may be right. I think I was trying to match a `#[cfg]` guard that was already in the code (around `make_linux_iflags()`, etc.) but it was probably unnecessary.